src/utils/system.py

```diff
@@ -1,35 +1,105 @@
-
-    You should have received a copy of the GNU Affero General Public License
-    along with this program; if not, see http://www.gnu.org/licenses
-
-"""
+#
+# Copyright (c) 2022 YunoHost Contributors
+#
+# This file is part of YunoHost (see https://yunohost.org)
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
 import re
 import os
 import logging
 
 from moulinette.utils.process import check_output
-from packaging import version
+from yunohost.utils.error import YunohostError
 
 logger = logging.getLogger("yunohost.utils.packages")
 
 YUNOHOST_PACKAGES = ["yunohost", "yunohost-admin", "moulinette", "ssowat"]
 
 
+def system_arch():
+    return check_output("dpkg --print-architecture")
+
+
+def system_virt():
+    """
+    Returns the output of systemd-detect-virt (so e.g. 'none' or 'lxc' or ...)
+    You can check the man of the command to have a list of possible outputs...
+    """
+    # Detect virt technology (if not bare metal) and arch
+    # Gotta have this "|| true" because it systemd-detect-virt return 'none'
+    # with an error code on bare metal ~.~
+    return check_output("systemd-detect-virt || true")
+
+
+def free_space_in_directory(dirpath):
+    stat = os.statvfs(dirpath)
+    return stat.f_frsize * stat.f_bavail
+
+
+def space_used_by_directory(dirpath, follow_symlinks=True):
+
+    if not follow_symlinks:
+        du_output = check_output(["du", "-sb", dirpath], shell=False)
+        return int(du_output.split()[0])
+
+    stat = os.statvfs(dirpath)
+    return stat.f_frsize * stat.f_blocks  # FIXME : this doesnt do what the function name suggest this does ...
+
+
+def human_to_binary(size: str) -> int:
+
+    symbols = ("K", "M", "G", "T", "P", "E", "Z", "Y")
+    factor = {}
+    for i, s in enumerate(symbols):
+        factor[s] = 1 << (i + 1) * 10
+
+    suffix = size[-1]
+    size = size[:-1]
+
+    if suffix not in symbols:
+        raise YunohostError(f"Invalid size suffix '{suffix}', expected one of {symbols}")
+
+    try:
+        size_ = float(size)
+    except Exception:
+        raise YunohostError(f"Failed to convert size {size} to float")
+
+    return int(size_ * factor[suffix])
+
+
+def binary_to_human(n: int) -> str:
+    """
+    Convert bytes or bits into human readable format with binary prefix
+    """
+    symbols = ("K", "M", "G", "T", "P", "E", "Z", "Y")
```

Comment:
The previous binary_to_human was supporting GiB notation. What should we do about that ?

Reply:
Is that really important x_x